openapiart/common.py

```diff
@@ -423,14 +423,25 @@ def validate_hex(self, hex):
     def validate_integer(self, value, min, max, type_format=None):
         if value is None or not isinstance(value, int):
             return False
-        if type_format == "uint32" and value < 0:
-            return False
-        if type_format == "uint64" and value < 0:
-            return False
         if min is not None and value < min:
             return False
         if max is not None and value > max:
             return False
+        if type_format is not None:
+            if type_format == "uint32" and (value < 0 or value > 4294967295):
+                return False
+            elif type_format == "uint64" and (
+                value < 0 or value > 18446744073709551615
+            ):
+                return False
+            elif type_format == "int32" and (
+                value < -2147483648 or value > 2147483647
+            ):
+                return False
+            elif type_format == "int32" and (
```

Comment:
should be "int64" (maybe add more tests since this wasn't caught ?)

Reply:
done